puppet/zulip/manifests/postgres_appdb_tuned.pp

```diff
@@ -80,6 +80,9 @@
   $ssl_key_file = zulipconf('postgresql', 'ssl_key_file', undef)
   $ssl_ca_file = zulipconf('postgresql', 'ssl_ca_file', undef)
 
+  # Only used in CentOS for now
+  $pg_datadir = "/var/lib/pgsql/${zulip::base::postgres_version}/data"
+
```

Comment:
Can you talk about how this commit was constructed?  Like, is the stuff at the bottom verbatim identical to the other postgres 10 one?

Reply:
Yes, copied verbatim from the other postgres 10.
Here is the diff between the 2 templates (left centos, right debian):
```
< data_directory = '<%= scope["zulip::postgres_appdb_tuned::pg_datadir"] %>'            # use data in another directory
---
> data_directory = '/var/lib/postgresql/<%= scope["zulip::base::postgres_version"] %>/main'             # use data in another directory
43c43
< hba_file = '<%= scope["zulip::postgres_appdb_tuned::pg_datadir"] %>/pg_hba.conf'      # host-based authentication file
---
> hba_file = '/etc/postgresql/<%= scope["zulip::base::postgres_version"] %>/main/pg_hba.conf'   # host-based authentication file
45c45
< ident_file = '<%= scope["zulip::postgres_appdb_tuned::pg_datadir"] %>/pg_ident.conf'  # ident configuration file
---
> ident_file = '/etc/postgresql/<%= scope["zulip::base::postgres_version"] %>/main/pg_ident.conf'       # ident configuration file
49c49
< external_pid_file = '/var/run/postgresql/<%= scope["zulip::base::postgres_version"] %>-main.pid'
---
> external_pid_file = '/var/run/postgresql/<%= scope["zulip::base::postgres_version"] %>-main.pid'                      # write an extra PID file
63,64c63,64
< #port = 5432                          # (change requires restart)
< #max_connections = 100                        # (change requires restart)
---
> port = 5432                           # (change requires restart)
> max_connections = 100                 # (change requires restart)
66c66
< #unix_socket_directories = '/tmp'     # comma-separated list of directories
---
> unix_socket_directories = '/var/run/postgresql'       # comma-separated list of directories
79c79
< #ssl = off
---
> ssl = on
84,85c84,85
< #ssl_cert_file = 'server.crt'
< #ssl_key_file = 'server.key'
---
> ssl_cert_file = '/etc/ssl/certs/ssl-cert-snakeoil.pem'
> ssl_key_file = '/etc/ssl/private/ssl-cert-snakeoil.key'
113c113
< #shared_buffers = 32MB                        # min 128kB
---
> shared_buffers = 128MB                        # min 128kB
127c127
< #dynamic_shared_memory_type = posix   # the default is the first option
---
> dynamic_shared_memory_type = posix    # the default is the first option
160c160
< #bgwriter_flush_after = 0             # measured in pages, 0 disables
---
> #bgwriter_flush_after = 512kB         # measured in pages, 0 disables
212c212
< #checkpoint_flush_after = 0           # measured in pages, 0 disables
---
> #checkpoint_flush_after = 256kB               # measured in pages, 0 disables
344c344
< log_destination = 'stderr'            # Valid values are combinations of
---
> #log_destination = 'stderr'           # Valid values are combinations of
350c350
< logging_collector = on                        # Enable capturing of stderr and csvlog
---
> #logging_collector = off              # Enable capturing of stderr and csvlog
356c356
< log_directory = 'log'                 # directory where log files are written,
---
> #log_directory = 'log'                        # directory where log files are written,
358c358
< log_filename = 'postgresql-%a.log'    # log file name pattern,
---
> #log_filename = 'postgresql-%Y-%m-%d_%H%M%S.log'      # log file name pattern,
362c362
< log_truncate_on_rotation = on         # If on, an existing log file with the
---
> #log_truncate_on_rotation = off               # If on, an existing log file with the
370c370
< log_rotation_age = 1d                 # Automatic rotation of logfiles will
---
> #log_rotation_age = 1d                        # Automatic rotation of logfiles will
372c372
< log_rotation_size = 0                 # Automatic rotation of logfiles will
---
> #log_rotation_size = 10MB             # Automatic rotation of logfiles will
445c445
< log_line_prefix = '%m [%p] '          # special values:
---
> log_line_prefix = '%m [%p] %q%u@%d '          # special values:
471c471
< #log_timezone = 'GMT'
---
> log_timezone = 'UTC'
476c476
< #cluster_name = '<%= scope["zulip::base::postgres_version"] %>/main'                  # added to process titles if nonempty
---
> cluster_name = '<%= scope["zulip::base::postgres_version"] %>/main'                   # added to process titles if nonempty
492c492
< stats_temp_directory = 'pg_stat_tmp'
---
> stats_temp_directory = '/var/run/postgresql/<%= scope["zulip::base::postgres_version"] %>-main.pg_stat_tmp'
565c565
< #datestyle = 'iso, mdy'
---
> datestyle = 'iso, mdy'
567c567
< #timezone = 'GMT'
---
> timezone = 'UTC'
580c580
< #lc_messages = 'C'                    # locale for system error message
---
> lc_messages = 'C.UTF-8'                       # locale for system error message
582,584c582,584
< #lc_monetary = 'C'                    # locale for monetary formatting
< #lc_numeric = 'C'                     # locale for number formatting
< #lc_time = 'C'                                # locale for time formatting
---
> lc_monetary = 'C.UTF-8'                       # locale for monetary formatting
> lc_numeric = 'C.UTF-8'                        # locale for number formatting
> lc_time = 'C.UTF-8'                           # locale for time formatting
587c587
< #default_text_search_config = 'pg_catalog.simple'
---
> default_text_search_config = 'pg_catalog.english'
647c647
< #include_dir = 'conf.d'                       # include files ending in '.conf' from
---
> include_dir = 'conf.d'                        # include files ending in '.conf' from
```

Reply:
For `data_directory` and friends, should we just switch those to use the new variables you added in the puppet manifests?  Then they'd be the same for both paths.

Also, why is `cluster_name` commented out?  Seems like that might be bad.

Reply:
Also there are a few other things, like `max_connections`, that I'm suspicious of.

Reply:
`cluster_name` is commented out because in its default template config version, it is commented out.

Reply:
It is not commented out in Debian template config, https://github.com/zulip/zulip/commit/964a1ac8a750c1d9b9547afe8cd630ff9f2ecc99#diff-ab70b2b375c0a3c87b09881eef193351R477.

Reply:
> For data_directory and friends, should we just switch those to use the new variables you added in the puppet manifests? Then they'd be the same for both paths.

The structure `pg_datadir` is repeated several times on CentOS, but the base directory is different for each of data directory, hba-file&ident-file on Debian.